rosgraph_monitor/src/monitor.cpp

```diff
@@ -212,6 +218,13 @@ void RosGraphMonitor::track_node_updates(
     auto [it, inserted] = nodes_.emplace(node_name, NodeTracking{});
     if (inserted) {
       RCLCPP_DEBUG(logger_, "New node: %s", node_name.c_str());
+      auto params = query_params_(node_name);
+      if (params.has_value()) {
+        it->second.params = *params;
+        RCLCPP_DEBUG(logger_, "Node %s has parameters: %zu", node_name.c_str(), params->size());
+      } else {
+        RCLCPP_DEBUG(logger_, "Node %s is unresponsive to parameter queries", node_name.c_str());
```

Comment:
I think we will need a retry mechanism. You may notice a node well before it can respond to service calls, for various reasons - potentially several seconds, if some node has really heavy startup for whatever reason. But we don't want to synchronously block on that, halting the rest of the operations, or fail to query it permanently.

I think some `std::async` (that can be cancelled when we notice node disappears), and a periodic check on the active future pool, would probably make sense.